File: 🟢 Easy/1518 - Water Bottles/_1518WaterBottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        maxUsedBottles = 0

        while numBottles >= numExchange:
            newFullBottles = numBottles // numExchange
            maxUsedBottles += newFullBottles * numExchange
            numBottles -= newFullBottles * numExchange
            numBottles += newFullBottles

        maxUsedBottles += numBottles
        return maxUsedBottles

File: 🟢 Easy/1518 - Water Bottles/_1518WaterBottles.c
int numWaterBottles(int numBottles, int numExchange) {
    int maxUsedBottles = 0;

    while (numBottles >= numExchange) {
        int newFullBottles = numBottles / numExchange;
        maxUsedBottles += newFullBottles * numExchange;
        numBottles -= newFullBottles * numExchange;
        numBottles += newFullBottles;
    }

    maxUsedBottles += numBottles;
    return maxUsedBottles;
}